src/index/base.cpp

```diff
@@ -61,12 +61,37 @@ class BaseIndexNotifications : public interfaces::Chain::Notifications
 
 void BaseIndexNotifications::blockConnected(const interfaces::BlockInfo& block)
 {
-    m_index.BlockConnected(block);
+    if (m_index.IgnoreBlockConnected(block)) return;
+
+    const CBlockIndex* pindex = &m_index.BlockIndex(block.hash);
+    const CBlockIndex* best_block_index = m_index.m_best_block_index.load();
+    if (block.chain_tip && best_block_index && best_block_index != pindex->pprev && !m_index.Rewind(best_block_index, pindex->pprev)) {
```

Comment:
In a2e87d9d6:

I asked myself why this `chain_tip` check was placed here few times until saw 5e2b1ea1, which makes `ThreadSync()` re-use this code.

Probably, it would be helpful for other reviewers to have a comment saying: "`chain_tip` is used here to let `ThreadSync()` handle reorgs".